pkg/bundle/create/create.go

```diff
@@ -154,17 +159,55 @@ func CreateDiff(configPath, rootDir string) error {
 		}
 	}
 
-	// Gather files we pulled
-	if err = bundle.ReconcileFiles(&metadata.MetadataSpec, rootDir); err != nil {
+	if err != nil {
 		return err
 	}
 
 	// Add mirror as a new PastMirror
 	lastRun.Sequence++
 	metadata.PastMirrors = append(metadata.PastMirrors, lastRun)
-	if err := config.WriteMetadata(metadata, rootDir); err != nil {
+
+	// Run archiver
+	if err := prepareArchive(sourceDir, outputDir, segSize, &metadata); err != nil {
+		return err
+	}
+
+	if err := config.WriteMetadata(metadata, rootAbs); err != nil {
 		return fmt.Errorf("error writing metadata: %v", err)
 	}
 
 	return nil
 }
+
+func prepareArchive(rootDir, outputDir string, segSize int64, metadata *v1alpha1.Metadata) error {
+	// Create archiver
+	arc := archive.NewArchiver()
+
+	// Set get absolute path to output dir
+	output, err := filepath.Abs(outputDir)
+
+	if err != nil {
+		return err
+	}
+
+	// Change dir before archving to avoid issues with symlink paths
+	if err := os.Chdir(rootDir); err != nil {
```

Comment:
Is it possible to archive without chdir, or at least `defer os.Chdir(oldWorkingDir)` so this function doesn't cause a potentially unexpected side-effect for downstream execution?

Reply:
@estroz All good with adding a `defer` statement to change back directories. Really good idea. The reason I am doing a `Chdir` is because of how the symlinks are being processed during the write. I believe there is an open PR on the archiver library to get that fixed.